PowerValidatedSolutions.psm1

```diff
@@ -50287,6 +50287,208 @@ Function Get-AriaNetworksDataSource {
+    <#
+        .SYNOPSIS
+        Add a new NSX-T Data Source to VMware Aria Operations for Networks.
+
+        .DESCRIPTION
+        The New-AriaNetworksNsxtDataSource cmdlet allows a user to add in a NSX-T Manager as a new data source in VMware Aria Operations for Networks in order to collect network flow data.
+
+        .EXAMPLE
+        New-AriaNetworksNsxtDataSource -fqdn sfo-m01-nsx01.sfo.rainpole.io -certificate F:\certs\sfo-m01-nsx01.cer -privatekey F:\certs\sfo-m01-nsx01.key -nickname "sfo-m01-nsx01 - Management Domain NSX Manager" -CollectorId 15832:901:1711011916294613031 -enabled true
+        This example adds a NSX-T Manager server as a new data source in VMware Aria Operations for Networks by using a NSXT-T principal identity user.
+
+        .EXAMPLE
+        New-AriaNetworksNsxtDataSource -fqdn sfo-m01-nsx01.sfo.rainpole.io -username svc-inv-vsphere -password VMw@re1!VMw@re1! -nickname "sfo-m01-nsx01 - Management Domain NSX Manager" -CollectorId 15832:901:1711011916294613031 -enabled true
+        This example adds a NSX-T Manager server as a new data source in VMware Aria Operations for Networks by using a NSX-T service account user with a password.
+
+        .PARAMETER fqdn
+        The NSX-T Manager to add as a data source.
+
+        .PARAMETER certificate
+        The principal identity certificate to use for authentication.
+
+        .PARAMETER privatekey
+        The principal identity private key to use for authentication.
+
+        .PARAMETER username
+        The username to use for authentication.
+
+        .PARAMETER password
+        The password to use for authentication.
+
+        .PARAMETER nickname
+        The nickname to use for this data source in VMware Aria Operations for Networks.
+
+        .PARAMETER collectorId
+        The id of the VMware Aria Operations for Networks collector node where the NSX-T Manager data source will be connected.
+
+        .PARAMETER enabled
+        The parameter to enable the datasource.
+
+    #>
+
+    [CmdletBinding(DefaultParametersetName = "credentials")][OutputType('System.Management.Automation.PSObject')]
+
+    Param (
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$fqdn,
+      [Parameter (Mandatory = $false, ParameterSetName = 'certificate')] [ValidateNotNullOrEmpty()] [String]$certificate,
+      [Parameter (Mandatory = $false, ParameterSetName = 'certificate')] [ValidateNotNullOrEmpty()] [String]$privatekey,
+      [Parameter (Mandatory = $false, ParameterSetName = 'credentials')] [ValidateNotNullOrEmpty()] [String]$username,
+      [Parameter (Mandatory = $false, ParameterSetName = 'credentials')] [ValidateNotNullOrEmpty()] [String]$password,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$nickname,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$collectorId,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$enabled
+    )
+
+    Try {
```

Comment:
Any reason your not consistently using the if ($ariaNetworksAppliance) { } else { Write-Error ""} consistently

Reply:
Resolved in https://github.com/vmware/power-validated-solutions-for-cloud-foundation/commit/5b6f2b46e0444235d6f3d74157fc959015d63688